### PR TITLE
Implements disabling_files

### LINKED ISSUE #31: Allow to disable Mergify if certain files are changed
This could be handy to disable Mergify for e.g. changes to `.mergify.yml` or any other very sensitive files that people would like to merge manually.

### PR DESCRIPTION
**Pull request by sileht**
Closes #31

### CODE REVIEW

File: doc/source/configuration.rst [hunk 1]
```diff
@@ -43,6 +43,12 @@ A ``Rule`` is a dictionary with the following optional keys:
      - A `label <https://help.github.com/articles/about-labels/>`_ name that
        will disable Mergify if it is applied to a pull request. Default is
        ``no-mergify``.
+   * - ``disabling_files``
+     - ``array of string``
+     - The list of files wildcard that will disable Mergify if
+       modified. Usually any files that configure Continuous Integration
+       workflow. Default is ``[.mergify.yml]``. Even if forgotten
+       ``.mergify.yml`` is always added.
```
Comment (jd) [inline doc/source/configuration.rst]: Rewrite with something like this:

```
A list of file paths that disables Mergify if they are part of the pull request. Any file that changes the behavior of the continuous integration workflow should be listed. The default is ``[.mergify.yml]``. Note that the engine always ignore pull requests that modify ``.mergify.yml`` anyway. Those pull requests will have to be merged manually, even if `.mergify.yml` is not listed in this setting.
```

File: mergify_engine/rules.py [hunk 2]
```diff
@@ -129,5 +130,9 @@ def get_branch_rule(g_repo, branch, ref=github.GithubObject.NotSet):
     except voluptuous.MultipleInvalid as e:
         raise NoRules("mergify configuration invalid: %s" % str(e))
 
+    # NOTE(sileht): Always disabled Mergify if it's configuration is changed
```
Comment (jd) [inline mergify_engine/rules.py]: disable

its

File: mergify_engine/gh_pr_fullifier.py [hunk 3]
```diff
@@ -160,11 +161,23 @@ def compute_travis_url(pull, **extra):
...
+def disabled_by_rules(pull, **extra):
     labels = [l.name for l in pull.labels]
     if extra["branch_rule"]["disabling_label"] in labels:
+        return "Disabled with label"
+
+    pull_files = [f.filename for f in pull.get_files()]
+    for w in extra["branch_rule"]["disabling_files"]:
+        filtered = fnmatch.filter(pull_files, w)
+        if filtered:
+            return "Disabled, %s have been changed" % filtered[0]
```
Comment (jd) [inline mergify_engine/gh_pr_fullifier.py]: `"Disabled, %s is modified in this pull request"`

sounds clearer if somebody does not know mergify.

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 34cc8eb66c28ed0e6d774a9b136d470503ac72ec

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 73fc66f5df34367c81d6467c689397618c6befbf

**Pull request merged**